src/equations/shallow_water_wet_dry_1d.jl

```diff
@@ -0,0 +1,677 @@
+or the entropy variables.
+This affects the implementation and use of these equations in various ways:
+* The flux values corresponding to the bottom topography must be zero.
+* The bottom topography values must be included when defining initial conditions, boundary conditions or
+  source terms.
+* [`AnalysisCallback`](@ref) analyzes this variable.
+* Trixi.jl's visualization tools will visualize the bottom topography by default.
+
+References for the SWE are many but a good introduction is available in Chapter 13 of the book:
+- Randall J. LeVeque (2002)
+  Finite Volume Methods for Hyperbolic Problems
+  [DOI: 10.1017/CBO9780511791253](https://doi.org/10.1017/CBO9780511791253)
+"""
+struct ShallowWaterEquationsWetDry1D{RealT <: Real} <:
+       Trixi.AbstractShallowWaterEquations{1, 3}
+    gravity::RealT # gravitational constant
+    H0::RealT      # constant "lake-at-rest" total water height
+    # `threshold_limiter` used in `PositivityPreservingLimiterShallowWater` on water height,
+    # as a (small) shift on the initial condition and cutoff before the next time step.
+    # Default is 500*eps() which in double precision is ≈1e-13.
+    threshold_limiter::RealT
+    # `threshold_wet` applied on water height to define when the flow is "wet"
+    # before calculating the numerical flux.
+    # Default is 5*eps() which in double precision is ≈1e-15.
+    threshold_wet::RealT
+end
+
+# Allow for flexibility to set the gravitational constant within an elixir depending on the
+# application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
+# The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
+# well-balancedness test cases.
+# Strict default values for thresholds that performed well in many numerical experiments
+function ShallowWaterEquationsWetDry1D(; gravity_constant, H0 = zero(gravity_constant),
+                                       threshold_limiter = nothing,
+                                       threshold_wet = nothing)
+    T = promote_type(typeof(gravity_constant), typeof(H0))
+    if threshold_limiter === nothing
+        threshold_limiter = 500 * eps(T)
+    end
+    if threshold_wet === nothing
+        threshold_wet = 5 * eps(T)
+    end
+    ShallowWaterEquationsWetDry1D(gravity_constant, H0, threshold_limiter,
+                                  threshold_wet)
+end
+
+Trixi.have_nonconservative_terms(::ShallowWaterEquationsWetDry1D) = True()
+function Trixi.varnames(::typeof(Trixi.cons2cons), ::ShallowWaterEquationsWetDry1D)
+    ("h", "h_v", "b")
+end
+# Note, we use the total water height, H = h + b, as the first primitive variable for easier
+# visualization and setting initial conditions
+function Trixi.varnames(::typeof(Trixi.cons2prim), ::ShallowWaterEquationsWetDry1D)
+    ("H", "v", "b")
+end
+
+# Set initial conditions at physical location `x` for time `t`
```

Comment:
It might be good to note that some of these initial conditions, source terms, boundary condition slip wall, etc.  correspond to fully wet testcases and functionality that already exist in Trixi.jl. However, we keep their operability here for convenience of development and debugging (or so). Also, the physical flux is same so we have a wrapper for it as well.

Reply:
I added a comment in 64b9bf37466ed8503b3f15f4ca63a2f28bbcd9b0